libtelnet.c

```diff
@@ -1297,6 +1354,45 @@ void telnet_send(telnet_t *telnet, const char *buffer,
 	}
 }
 
+/* send non-command text (escapes IAC bytes and does NVT translation) */
+void telnet_send_text(telnet_t *telnet, const char *buffer,
+		size_t size) {
+	size_t i, l;
+
+	for (l = i = 0; i != size; ++i) {
+		/* dump prior portion of text, send escaped bytes */
+		if (buffer[i] == (char)TELNET_IAC) {
+			/* dump prior text if any */
+			if (i != l) {
+				_send(telnet, buffer + l, i - l);
+			}
+			l = i + 1;
+
+			/* send escape */
+			telnet_iac(telnet, TELNET_IAC);
+		}
+		/* special characters if not in BINARY mode */
+		else if (!(telnet->flags & TELNET_FLAG_TRANSMIT_BINARY) &&
+				 (buffer[i] == '\r' || buffer[i] == '\n')) {
+			/* dump prior portion of text */
+			if (i != l)
+				_send(telnet, buffer + l, i - l);
```

Comment:
OK.  I generally prefer to omit braces where they are not needed, but try to follow conventions in other people's code.  I noted that there are statement both with and without in libtelnet.  My guess as to how these two cases became inconsistent is that I copied 1379 with the braces but then later removed them after writing the next if statement at 1384 without braces and then noticing that 1379 was inconsistent with 1384.  I'd be happy to put braces on both.